### PR TITLE
Add weighted features

### PR DESCRIPTION
**Pull request by cjauvin**
Hi Liang Sun, sorry for the back and forth! This is a version with the minimal set of changes needed to implement the weighted features.

### CODE REVIEW

File: simhash/__init__.py [hunk 1]
```diff
@@ -69,12 +69,25 @@ def build_by_text(self, content):
...
+                   a token -> weight dict.
+        """
         v = [0] * self.f
         masks = [1 << i for i in range(self.f)]
-        for h in hashs:
+        if isinstance(features, dict):
+            features = features.items()
+        for f in features:
+            if isinstance(f, collections.Iterable) and \
+               not isinstance(f, basestring):
```
Comment (liangsun) [inline simhash/__init__.py]: Why not just check if `f` is `basestring`? What's the reason to check if `f` is `collections.Iterable`?

File: tests/test_simhash.py [hunk 2]
```diff
@@ -42,6 +42,32 @@ def test_short(self):
...
+        # test features as lists of (token, weight) tuples
+        for i in range(D.shape[0]):
+            Di = D.getrow(i)
+            features = zip([voc[j] for j in Di.indices], Di.data)
+            self.assertNotEqual(Simhash(features).value, 0)
+        # test features as token -> weight dicts
+        for i in range(D.shape[0]):
+            Di = D.getrow(i)
+            features = dict(zip([voc[j] for j in Di.indices], Di.data))
+            self.assertNotEqual(Simhash(features).value, 0)
```
Comment (liangsun) [inline tests/test_simhash.py]: Test cases are too simple and can only cover little situations.

File: tests/test_simhash.py [hunk 3]
```diff
@@ -42,6 +42,32 @@ def test_short(self):
...
+
```
Comment (liangsun) [inline tests/test_simhash.py]: Would you add more test cases? Currently these two cases only cover a small scope.

Comment (cjauvin) [inline tests/test_simhash.py]: I agree the testing as it is is not terribly exhaustive. Before I spend too much time trying to come up with clever tests (which is not exactly trivial for such a thing), would you tell me what kind of testing would satisfy you?

Comment (liangsun) [inline tests/test_simhash.py]: @cjauvin Some non-trivial cases are Ok, like the distance between `data[0]` and `data[1]` is less than the distance between `data[1]` and `data[2]`.

**Comment by cjauvin** [comment]
@liangsun Please tell me what you think of these additional tests.

**Comment by liangsun** [comment]
@cjauvin :+1: :beers:

**Pull request merged**